scripts/generate_specialist_skill_matrix.py

```
@@ -0,0 +1,136 @@
+
+def main() -> int:
+    ap = argparse.ArgumentParser()
+    ap.add_argument("--check", action="store_true", help="fail if the doc is stale")
+    args = ap.parse_args()
+
+    content = render_markdown(build_matrix())
+    if args.check:
+        existing = _DOC_PATH.read_text(encoding="utf-8") if _DOC_PATH.exists() else ""
+        # Compare ignoring the volatile "Last generated" line.
+        def _strip_ts(s: str) -> str:
+            return "\n".join(l for l in s.splitlines() if not l.startswith("> Last generated:"))
```

Ruff (0.15.15)

[error] Ambiguous variable name: `l`

(E741)